### PR TITLE
Sanjam/TS Migration Inline Icon with Tests

### PR DESCRIPTION
**Pull request by sanjam-deriv**
## Changes:

Please include a summary of the change and which issue is fixed below:
-   ts migration of inline note with icon and tests

## When you need to add unit test

-   [ ] If this change disrupt current flow
-   [ ] If this change is adding new flow

## When you need to add integration test

-   [ ] If components from external libraries are being used to define the flow, e.g. @deriv/components
-   [ ] If it relies on a very specific set of props with no default behavior for the current component.

## Test coverage checklist (for reviewer)

-   [ ] Ensure utility / function has a test case 
-   [ ] Ensure all the tests are passing

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: packages/account/src/Components/inline-note-with-icon/inline-note-with-icon.tsx [hunk 1]
```diff
@@ -1,8 +1,16 @@
...
 import { Icon, Text } from '@deriv/components';
 
-const InlineNoteWithIcon = ({ icon, message, title }) => {
+type TInlineNoteWithIconExtend = {
+    icon: string;
+    message: string;
+    title: string;
+};
+
+export type TInlineNoteWithIcon = HTMLAttributes<HTMLInputElement | HTMLLabelElement> &
```
Comment (iman-fs) [inline packages/account/src/Components/inline-note-with-icon/inline-note-with-icon.tsx:10]: just for my information, why did you add `HTMLAttributes<HTMLInputElement | HTMLLabelElement>` ?because I checked and without having it there is no error

File: packages/account/src/Components/inline-note-with-icon/inline-note-with-icon.tsx [hunk 2]
```diff
@@ -1,8 +1,15 @@
 import React from 'react';
-import PropTypes from 'prop-types';
 import { Icon, Text } from '@deriv/components';
 
-const InlineNoteWithIcon = ({ icon, message, title }) => {
+type TInlineNoteWithIconExtend = {
+    icon: string;
+    message: string;
```
Comment (amina-deriv) [inline packages/account/src/Components/inline-note-with-icon/inline-note-with-icon.tsx:6]: Can't this message and string copme with <Localize/>  component? Then it is of type ReactNode. right?

File: packages/account/src/Components/inline-note-with-icon/inline-note-with-icon.tsx [hunk 3]
```diff
@@ -1,8 +1,15 @@
...
+    title: string;
+};
+
+export type TInlineNoteWithIcon = React.PropsWithChildren<TInlineNoteWithIconExtend>;
+
+const InlineNoteWithIcon = ({ icon, message, title }: TInlineNoteWithIcon) => {
```
Comment (amina-deriv) [inline packages/account/src/Components/inline-note-with-icon/inline-note-with-icon.tsx:13]: pls check this, we can just use TInlineNoteWithIconExtend as the prop type. no need to extend  props here

File: packages/account/src/Components/inline-note-with-icon/__tests__/inline-note-with-icon.spec.tsx [hunk 4]
```diff
@@ -1,6 +1,6 @@
 import React from 'react';
 import { screen, render } from '@testing-library/react';
-import InlineNoteWithIcon from '../inline-note-with-icon';
+import InlineNoteWithIconExtend from '../inline-note-with-icon';
```
Comment (amina-deriv) [inline packages/account/src/Components/inline-note-with-icon/__tests__/inline-note-with-icon.spec.tsx:3]: InlineNoteWithIconExtend is the type name. right?

**Review by amina-deriv — APPROVED** [review]

**Review by likhith-deriv — APPROVED** [review]

File: packages/account/src/Components/inline-note-with-icon/inline-note-with-icon.tsx [hunk 5]
```diff
@@ -1,8 +1,13 @@
 import React from 'react';
-import PropTypes from 'prop-types';
 import { Icon, Text } from '@deriv/components';
 
-const InlineNoteWithIcon = ({ icon, message, title }) => {
+export type TInlineNoteWithIconExtend = {
```
Comment (iman-fs) [inline packages/account/src/Components/inline-note-with-icon/inline-note-with-icon.tsx:4]: Not sure why we need to export this type but if you need to use it somewhere else ( where `<InlineNoteWithIcon/>` is also being used ) you can use `React.ComponentProps<typeof InlineNoteWithIcon> ` . it will give you access to `TInlineNoteWithIconExtend` type and if you need some specific type in it you can grab them using `Pick<type>` utility.
this way you don't need to import the type along with the component

**Review by iman-fs — APPROVED** [review]
